### PR TITLE
Register the service when the component state is healthy

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: Cai.Zhang <cai.zhang@zilliz.com>

issue: #13072

### CODE REVIEW

File: internal/distributed/rootcoord/service.go [hunk 1]
```diff
@@ -258,6 +253,11 @@ func (s *Server) startGrpcLoop(port int) {
 func (s *Server) start() error {
 	log.Debug("RootCoord Core start ...")
 	if err := s.rootCoord.Start(); err != nil {
+		log.Error(err.Error())
```
Comment (congqixia) [inline internal/distributed/rootcoord/service.go:256]: log format can be refine in the future

File: internal/distributed/indexcoord/service.go [hunk 2]
```diff
@@ -109,6 +105,11 @@ func (s *Server) start() error {
 		return err
 	}
 	log.Debug("indexCoord started")
+	if err := s.indexcoord.Register(); err != nil {
+		log.Error("IndexCoord", zap.Any("register session error", err))
```
Comment (congqixia) [inline internal/distributed/indexcoord/service.go:109]: Error message should be like "Index Register failed", with zap.Error(err)

File: internal/distributed/indexcoord/service.go [hunk 3]
```diff
@@ -100,6 +95,7 @@ func (s *Server) init() error {
 		log.Error("IndexCoord", zap.Any("init error", err))
```
Comment (congqixia) [inline internal/distributed/indexcoord/service.go:95]: IndexCoord start grpc failed, zap.Error(err)

File: internal/indexnode/indexnode.go [hunk 4]
```diff
@@ -113,13 +113,17 @@ func NewIndexNode(ctx context.Context) (*IndexNode, error) {
...
 // Register register index node at etcd.
 func (i *IndexNode) Register() error {
-	i.session = sessionutil.NewSession(i.loopCtx, Params.MetaRootPath, Params.EtcdEndpoints)
-	if i.session == nil {
-		return errors.New("failed to initialize session")
-	}
-	i.session.Init(typeutil.IndexNodeRole, Params.IP+":"+strconv.Itoa(Params.Port), false)
-	Params.NodeID = i.session.ServerID
-	Params.SetLogger(Params.NodeID)
+	i.session.Register()
```
Comment (congqixia) [inline internal/indexnode/indexnode.go:116]: return error shall be checked

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/distributed/rootcoord/service.go (hunk 1 above)
Comment (xiaocai2333) [reply internal/distributed/rootcoord/service.go:256]: Yeah, I will sort out the startup logic of each component, and sort out these logs by the way.

File: internal/distributed/indexcoord/service.go (hunk 2 above)
Comment (xiaocai2333) [reply internal/distributed/indexcoord/service.go:109]: Same as above.

File: internal/distributed/indexcoord/service.go (hunk 3 above)
Comment (xiaocai2333) [reply internal/distributed/indexcoord/service.go:95]: Same as above.

**Pull request merged**